File: 1769_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* minOperations(char* boxes, int* returnSize) {
    int n = strlen(boxes);
    int* result = (int*)calloc(n, sizeof(int));

    int balls = 0, ops = 0;
    for (int i = 0; i < n; i++) {
        result[i] += ops;
        balls += (boxes[i] == '1');
        ops += balls;
    }

    balls = ops = 0;
    for (int i = n - 1; i >= 0; i--) {
        result[i] += ops;
        balls += (boxes[i] == '1');
        ops += balls;
    }

    *returnSize = n;
    return result;
}

File: 1769_Minimum_Number_of_Operations_to_Move_All_Balls_to_Each_Box/solution.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        n = len(boxes)
        result = [0] * n

        balls = ops = 0
        for i in range(n):
            result[i] += ops
            balls += boxes[i] == "1"
            ops += balls

        balls = ops = 0
        for i in range(n - 1, -1, -1):
            result[i] += ops
            balls += boxes[i] == "1"
            ops += balls

        return result
